### PR TITLE
feat: rate limit removeMember

### PR DESCRIPTION
**Pull request by sean-brydon**
adds sourceIP to ctx + rate limit remove member

### CODE REVIEW

File: packages/trpc/server/createContext.ts [hunk 1]
```diff
@@ -12,6 +13,7 @@ import type { CreateNextContextOptions } from "@trpc/server/adapters/next";
 type CreateContextOptions = CreateNextContextOptions | GetServerSidePropsContext;
 
 export type CreateInnerContextOptions = {
+  sourceIp?: string;
```
Comment (sean-brydon) [inline packages/trpc/server/createContext.ts:16]: Maybe @zomars @keithwillcode can figure out a better way to have this not be optional 

It needs to be optional due to the fact CreateSsgHelpers needs the full context - however we have no way of getting the IP into that. they also use the infered types so we can't omit in one place and not the other.

File: packages/trpc/server/routers/viewer/teams/removeMember.handler.ts [hunk 2]
```diff
@@ -14,11 +15,16 @@ type RemoveMemberOptions = {
   ctx: {
     user: NonNullable<TrpcSessionUser>;
     prisma: PrismaClient;
+    sourceIp?: string;
```
Comment (sean-brydon) [inline packages/trpc/server/routers/viewer/teams/removeMember.handler.ts:18]: This is always there - check getIP return signature. Type issue within context i cant seem to figure out

**Review by zomars — COMMENTED** [review]
Are we experiencing abuse on this endpoint?

**Review by ThyMinimalDev — APPROVED** [review]
looking good, can't figure out the type issue, approving as is

**Pull request merged**